Consistent dialog's and removing the use of deprecated MaterialDialog.Builder.

asaikarthikeya opened the pull request:
**Total **13** dialog's are converted to Material AlertDialogBuilder.**

Screenshots of a few dialog's - 

**Other Section** 

2 dialog's - Logout + Close envirocar app.

<img src="https://user-images.githubusercontent.com/70392921/120886787-55208080-c60d-11eb-96f7-7ab524e5fc63.jpg" width="225" height="450" />           <img src="https://user-images.githubusercontent.com/70392921/120886788-5651ad80-c60d-11eb-989e-61a74e5bee0e.jpg" width="225" height="450" />


**Dashboard** 

Logout in dashboard 

<img src="https://user-images.githubusercontent.com/70392921/120886821-839e5b80-c60d-11eb-9285-ee1b18cd6d22.jpg" width="225" height="450" />


**Carselection** 

Delete car dialog 

<img src="https://user-images.githubusercontent.com/70392921/120886841-9b75df80-c60d-11eb-899b-b007a9df06ed.jpg" width="225" height="450" />



**OBD Adapter** 

2 dialog's - bluetooth pairing , delete pairing.

<img src="https://user-images.githubusercontent.com/70392921/120886900-db3cc700-c60d-11eb-833a-20a795d94856.jpg" width="225" height="450" /> <img src="https://user-images.githubusercontent.com/70392921/120886903-e0017b00-c60d-11eb-8af3-477dfe7c99e9.jpg" width="225" height="450" />

asaikarthikeya commented:
Need to merge other 2 PR's in order to complete this.

asaikarthikeya commented:
@SebaDro  Used  [MaterialAlertDialogBuilder ](https://developer.android.com/reference/com/google/android/material/dialog/MaterialAlertDialogBuilder) to replace the existing Alertdialog.builder and Materialdialog.builder.

asaikarthikeya commented:
@SebaDro Please review the changes sir. Thank you ;)

asaikarthikeya commented:
Updated changes to all dialog's: No backgroundTint for dialog buttons.

<img src="https://user-images.githubusercontent.com/70392921/122261372-f5ba4e80-cef1-11eb-9617-30f6ce18be82.jpg" width="225" height="450" />

SebaDro reviewed (changes requested):
Let me have some remarks on your approach for introducing consistent dialogs:
1. There is no need to introduce the `MaterialAlertDialogBuilder` when setting a custom view on it. It could have been implemented using the `AlertDialog.Builder`, too. Though, I woulf prefer the  `MaterialAlertDialogBuilder`, but by using a customized theme without setting the view. Maybe, you find a way to adapt the theming style by enhancing the style:
```
<style name="MaterialDialog" parent="@style/ThemeOverlay.MaterialComponents.MaterialAlertDialog">
        <item name="buttonBarPositiveButtonStyle">@style/PositiveButtonStyle</item>
        <item name="buttonBarNegativeButtonStyle">@style/NegativeButtonStyle</item
</style>
``` 

2. I found that [DialogUtils class](https://github.com/enviroCar/enviroCar-app/blob/gsoc2021/org.envirocar.app/src/org/envirocar/app/views/utils/DialogUtils.java), which provides the deprecated `MaterialDialog.Builder`, is still used in some activities. I would suggest to replace those lines by the `MaterialAlertDialogBuilder`.

3. If it is not possible to theme the `MaterialAlertDialogBuilder` like described under 1., you should consider introducing a utility class for constructing the dialogs usingthe `MaterialAlertDialogBuilder`, like it's done in the `DialogUtils` class. That would make a lot of boilerplate code for inflating the view within the activity classes obsolete.

asaikarthikeya commented on org.envirocar.app/src/org/envirocar/app/views/logbook/LogbookActivity.java:
```diff
@@ -144,7 +144,8 @@ public boolean onItemLongClick(AdapterView<?> parent, View view, int position, l
                         .genral_dialog_toolbar);
                 toolbar1.setTitle(R.string.logbook_dialog_delete_fueling_header);
-                toolbar1.setNavigationIcon(ContextCompat.getDrawable(LogbookActivity.this, R.drawable.others_logbook_white_24dp));
+                toolbar1.setNavigationIcon(ContextCompat.getDrawable(LogbookActivity.this,
```
Updated this yesterday - 
<img src="https://user-images.githubusercontent.com/70392921/122352891-25f30300-cf6d-11eb-9b67-82486c167794.jpg" width="225" height="450" />

asaikarthikeya commented:
@SebaDro 
1. I have removed the custom view and set the dialog style in `style.xml`. 
2. Converted the dialog's in DialogUtils class to `MaterialAlertDialogBuilder`. 
3. Deleted the view files of Bluetooth , dialog util and general dialog view that i created.
3. Improved the dialogs that were implemented using this DialogUtils class -- Sampling rate and Timepicker .prefrence dialog's.

Refer the below screenshots - Improved dialogUtils dialogs ( Old - Improved)
<img src="https://user-images.githubusercontent.com/70392921/122459864-13111a80-cfcf-11eb-97d4-2bac5be31690.jpg" width="225" height="450" /> <img src="https://user-images.githubusercontent.com/70392921/122458294-494d9a80-cfcd-11eb-8e96-c7fd75d736ab.jpg" width="225" height="450" /> 

Other dialog's after removing custom view and updating from styles.

<img src="https://user-images.githubusercontent.com/70392921/122458299-4a7ec780-cfcd-11eb-8d80-1af89be91486.jpg" width="225" height="450" />

SebaDro commented on org.envirocar.app/src/org/envirocar/app/views/utils/DialogUtils.java:
```diff
@@ -36,46 +36,34 @@
     }
 
-    public static MaterialDialog.Builder createDefaultDialogBuilder(
+    public static MaterialAlertDialogBuilder createDefaultDialogBuilder(
```
Replacing `MaterialDialog.Builder` by `MaterialAlertDialogBuilder` here is noit enough. Have a look in those activities that makes use of the `DialogUtils` class. You will notice, that those dialogs are broke, since your custom style as well as title, message and icon are not set.

SebaDro reviewed (changes requested):
Great to see that you have implemented an approach that makes fully advantage of the MaterialAlertDialog by creating a custom theme for it 😃.

Now, there are still some artifacts left, that effects in broken dialogs because `DialogUtils` is still used in a few activities. E.g., the dialog for adding a car, only shows the two buttons:

<img src="https://user-images.githubusercontent.com/18207870/122540165-a6c30500-d028-11eb-9db6-bc1b6de1303f.jpg" width="225" height="450" />

Since layout of the dialogs now is created by your custom theme, there is no more need to have a utility class for constructing the dialogs' layout manually. Therefore, I suggest some last refinements:
1. Revert your changes inside the `DialogUtils` class and mark it as deprecated.
2. Identify those activities that still makes use of the  `DialogUtils` class
3. In those activities, replace the call of `DialogUtils.createDefaultDialogBuilder()` by creating and showing the dialogs via `MaterialAlertDialogBuilder`, such as you have done in all othe actvities.

asaikarthikeya commented:
<img src="https://user-images.githubusercontent.com/70392921/122564081-962f8080-d062-11eb-8fd5-119f2e9f62ec.jpg" width="225" height="450" />

asaikarthikeya commented on org.envirocar.app/src/org/envirocar/app/views/login/SignupActivity.java:
```diff
@@ -226,16 +231,16 @@ private void register(String username, String email, String password) {
-                            .cancelable(false)
-                            .positiveText(R.string.ok)
-                            .cancelListener(dialog1 -> {
+                    final AlertDialog d = new MaterialAlertDialogBuilder(SignupActivity.this,R.style.MaterialDialog)
```
Done :)
